samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb

```diff
@@ -421,9 +487,75 @@
+      "2018-11-26 19:29:23:INFO:10 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:28:INFO:15 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:33:INFO:20 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:38:INFO:25 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:43:INFO:30 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:48:INFO:35 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:53:INFO:40 seconds: waiting for job to complete\n",
+      "2018-11-26 19:29:58:INFO:45 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:03:INFO:50 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:08:INFO:55 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:13:INFO:60 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:18:INFO:65 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:23:INFO:70 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:28:INFO:75 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:33:INFO:80 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:38:INFO:85 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:43:INFO:90 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:48:INFO:95 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:53:INFO:100 seconds: waiting for job to complete\n",
+      "2018-11-26 19:30:58:INFO:105 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:03:INFO:110 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:08:INFO:115 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:13:INFO:120 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:18:INFO:125 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:24:INFO:130 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:29:INFO:135 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:34:INFO:140 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:39:INFO:145 seconds: waiting for job to complete\n",
+      "2018-11-26 19:31:39:INFO:Kaniko job complete.\n",
+      "2018-11-26 19:31:39:INFO:Build component complete.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from kfp import compiler\n",
     "\n",
-    "Now that we've tested the function locally, we want to build a component that can run as a step in the pipeline. First we need to build a base docker container image. We need TensorFlow and google-api-python-client packages."
+    "# The return value \"DeployerOp\" represents a step that can be used directly in a pipeline function\n",
+    "DeployerOp = compiler.build_python_component(\n",
+    "    component_func=deploy_model,\n",
+    "    staging_gcs_path=OUTPUT_DIR,\n",
+    "    dependency=[kfp.compiler.VersionedDependency(name='google-api-python-client', version='1.7.0')],\n",
```

Comment:
Not related to this change. Can we do:

dependency=['google-api-python-client==1.7.0', 'numpy>=1.2', 'tensorflow']

instead if having a "VersionedDependency" wrapper?

Reply:
I understand. It should be a separate change.

I am not worried about user not knowing detailed format. It is a well known format for Python users, and there are good docs to point them to (setup.py, requirements.txt, etc). For us, we just need to take the strings, maybe do some simple validations, and then pass them to Python's world. I feel that reduces the surface area. To me, it is always nice to align to a certain accepted convention. Let's also chat offline about this.

Reply:
Sure. Let's sync offline.